src/Campaigns/Models/Campaign.php

```diff
@@ -138,6 +138,24 @@ public function save(): void
         }
     }
 
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function addForm(DonationForm $donationForm, $updateDefaultDonationForm = false)
+    {
+        if ( ! $this->id) {
+            give(CampaignRepository::class)->insert($this, $donationForm);
+        } else {
+            if ( ! $this->defaultForm() && ! $updateDefaultDonationForm) {
+                $updateDefaultDonationForm = true;
+            }
+
+            give(CampaignRepository::class)->update($this, $donationForm, $updateDefaultDonationForm);
```

Comment:
So instead of doing all these and modifying Campaign model `insert` and `update` methods, I will add a new method to `CampaignRepository` - `addCampaignForm` which will take three parameters (Campaign, $donationFormId, $isDefault = false), and then handle adding form there.

Reply:
Resolved in 0714b47594199a3bb112162fbf5e9bfcb0e8d6f4